Makefile.reconcilermanager

```diff
@@ -44,21 +45,27 @@ configsync-crds: install-controller-gen "$(KUSTOMIZE)" "$(GOBIN)/addlicense"
 	&& rm ./manifests/configmanagement.gke.io_namespaceconfigs.yaml \
 	&& rm ./manifests/configmanagement.gke.io_repoes.yaml \
 	&& rm ./manifests/configmanagement.gke.io_syncs.yaml \
-	&& "$(GOBIN)/addlicense" ./manifests
+	&& "$(ADDLICENSE)" ./manifests
+
+"$(CONTROLLER_GEN)": buildenv-dirs
+	GOPATH="$(GO_DIR)" go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION)
 
 .PHONY: install-controller-gen
-# install controller-gen from source
-install-controller-gen:
-	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.13.0
+# install controller-gen (user-friendly target alias)
+install-controller-gen: "$(CONTROLLER_GEN)"
+
+.PHONY: clean-controller-gen
+clean-controller-gen:
+	@rm -rf $(CONTROLLER_GEN)
 
 .PHONY: generate-in-docker
 # Run make generate-in-docker in the docker buildenv container
-generate-in-docker: buildenv-dirs
+generate-in-docker: "$(CONTROLLER_GEN)"
 	@docker run $(DOCKER_RUN_ARGS) \
 		make generate
 
 .PHONY: configsync-crds-in-docker
 # Run make configsync-crds in the docker buildenv container
-configsync-crds-in-docker: buildenv-dirs
+configsync-crds-in-docker: "$(CONTROLLER_GEN)" "$(KUSTOMIZE)" "$(ADDLICENSE)"
```

Comment:
I don't think this needs `"$(CONTROLLER_GEN)" "$(KUSTOMIZE)" "$(ADDLICENSE)"` as dependencies, since those are covered in `configsync-crds`

ditto for `generate-in-docker` above

I think for these commands, we explicitly want to use the go binary inside the container to install each tool for portability

Reply:
I added them explicitly to make sure they were installed by the local user. But I guess DOCKER_RUN_ARGS does set the UID, so it shouldn't matter.

Reply:
reverted